src/uma_crypto.udl

```diff
@@ -13,14 +13,31 @@ namespace uma_crypto {
 
     [Throws=CryptoError]
     KeyPair generate_keypair();
+
+    [Throws=Bech32Error]
+    string encode_bech32(string hrp, sequence<u8> message_data);
+
+    [Throws=Bech32Error]
+    Bech32Data decode_bech32(string bech32_str);
 };
 
 interface KeyPair {
     sequence<u8> get_public_key();
     sequence<u8> get_private_key();
 };
 
+interface Bech32Data {
```

Comment:
You can use dictionary for this type instead of interface. Then you won't need to add the getter functions. 

https://mozilla.github.io/uniffi-rs/latest/udl/structs.html

Reply:
sg.  I did that initially but followed the pattern of the KeyPair, which does the impl + struct . Didn't know if that was a common convention in Rust